blocks/bricks/wrapper.py

```diff
@@ -0,0 +1,36 @@
+from theano import tensor
+
+from blocks.bricks.base import Brick, application
+
+
+class As2D(Brick):
+    """Wraps a 2D application around reshapes to accomodate N-D tensors.
+
+    The input is first flattened across all but its last axis, then the
+    application method is called on the flattened input, and finally
+    the output is reshaped so that it has the same number of dimensions
+    as the input.
+
+    Parameters
+    ----------
+    application_method : callable
+        Some brick's :meth:`apply` method
+
+    """
+    def __init__(self, application_method, **kwargs):
+        super(As2D, self).__init__(**kwargs)
+        self.application_method = application_method
+        self.children = [self.application_method.brick]
+
+    @application(inputs=['input_'], outputs=['output'])
```

Comment:
When a brick is one input and one output they are typically never used. So I would add it later if it turns out necessary.